src/config.rs

```diff
@@ -269,23 +269,59 @@ pub struct WindowConfig {
     dimensions: Dimensions,
 
     /// Pixel padding
-    #[serde(default="default_padding", deserialize_with = "deserialize_padding")]
-    padding: Delta<u8>,
+    #[serde(default, deserialize_with = "deserialize_padding")]
+    padding: Padding,
 
     /// Draw the window with title bar / borders
     #[serde(default, deserialize_with = "failure_default")]
     decorations: bool,
 }
 
-fn default_padding() -> Delta<u8> {
-    Delta { x: 2, y: 2 }
+fn default_padding() -> Padding {
+    Padding {
+        top: default_padding_field(),
+        right: default_padding_field(),
+        bottom: default_padding_field(),
+        left: default_padding_field(),
+        x: default_padding_field_deprecated(),
+        y: default_padding_field_deprecated(),
+    }
+}
+
+fn default_padding_field() -> u8 {
+    2
+}
+
+fn default_padding_field_deprecated() -> Option<u8> {
+    None
 }
 
-fn deserialize_padding<'a, D>(deserializer: D) -> ::std::result::Result<Delta<u8>, D::Error>
+fn deserialize_padding<'a, D>(deserializer: D) -> ::std::result::Result<Padding, D::Error>
     where D: de::Deserializer<'a>
 {
-    match Delta::deserialize(deserializer) {
-        Ok(delta) => Ok(delta),
+    use ::util::fmt;
+    match Padding::deserialize(deserializer) {
+        Ok(mut padding) => {
+            match padding.x {
+                Some(x) => {
+                    padding.left = x;
+                    padding.right = x;
+                        eprintln!("{}", fmt::Yellow("Config `padding.x` is deprecated. Please use \
+                                                    `right|left` instead"));
+                },
+                None => {}
+            }
+            match padding.y {
+                Some(y) => {
+                    padding.top = y;
+                    padding.bottom = y;
+                    eprintln!("{}", fmt::Yellow("Config `padding.y` is deprecated. Please use \
+                                                `top|bottom` instead"));
+                },
+                None => {}
```

Comment:
Instead of doing `match padding.y { Some(y) => { ... }, None => {} }` it would be more ergonomic to use `if let Some(y) => { ... }`. Like this the `None => {}` part which does nothing can just be left out.

Reply:
Ah yes that's much nicer. I first tried with `if padding.y.is_some()`, but that involved an `unwrap`, `if let` was exactly what I was looking for, thanks!